pulp_smash/tests/rpm/api_v2/test_unavailable_checksum.py

```diff
@@ -146,3 +173,46 @@ def verify_primary_xml(self, cfg, distributor, checksum_type):
             package.find(xpath).get('type') for package in packages
         }
         self.assertEqual(checksum_types, {checksum_type})
+
+    def verify_filelists_xml(self, cfg, distributor, checksum_type):
+        """Verify a published repo's filelists.xml uses the given checksum."""
+        filelist_xml = get_repodata(cfg, distributor, 'filelists')
+        xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/filelists'])
+        packages = filelist_xml.findall(xpath)
+        pkgids_len = {len(package.get('pkgid')) for package in packages}
+        self.verify_pkgid_len(
+            pkgids_len, checksum_type, 'verify_filelists.xml')
+
+    def verify_other_xml(self, cfg, distributor, checksum_type):
+        """Verify a published repo's other.xml uses the given checksum."""
+        other_xml = get_repodata(cfg, distributor, 'other')
+        xpath = '{{{}}}package'.format(RPM_NAMESPACES['metadata/other'])
+        packages = other_xml.findall(xpath)
+        pkgids_len = {len(package.get('pkgid')) for package in packages}
+        self.verify_pkgid_len(pkgids_len, checksum_type, 'verify_other.xml')
+
+    def verify_pkgid_len(self, pkgid_len, checksum_type, comment):
+        """Perform assertion based on given the checksum type.
+
+        Based on the checksum type provided different assertions will
+        be performed over the length of pkgid.
+        """
+        if checksum_type == 'sha256':
+            # sha256 is 256 bit long, and uses 4 bits
+            # then 256/4 = 64
+            with self.subTest(comment=comment):
+                self.assertEqual(pkgid_len, {64})
+        if checksum_type == 'sha1':
+            # sha1 is 160 bit long, and uses 4 bits
+            # then 160/4 = 40
+            with self.subTest(comment=comment):
```

Comment:
How about dropping this `subTest` context manager for the reason outlined above?

Reply:
Updated per comments.

@Ichimonji10 Thanks for the review.